### ISSUE TITLE
Single Types: Localization should appear as extra pages

### ISSUE THREAD

**Issue by byronwilliams** [strapi/gatsby-source-strapi#287]
The localizations for Single Types do not get expanded out to have their own Gatsby Page ID. This makes it challenging to use the Localization of a Single Type as a Page.

Example Config

![Screenshot from 2022-02-17 08-30-14](https://user-images.githubusercontent.com/303604/154436576-e76bc4b0-d1dc-4e8b-85ff-7d8e463240a4.png)

Example GraphQL 

![image](https://user-images.githubusercontent.com/303604/154436776-bb27ab74-c845-4daa-96cc-6c81e3d88d30.png)

Expected behavior:

The GraphQL result would look like:

```json
{
  "data": {
    "allStrapiSiteConfig": {
      "nodes": [
        {
          "locale": "en-GB",
          "hreflang": "en-gb",
          "enabled": true,
          "id": "3369956c-7e5b-5197-96cf-52a744bf9a90"
        },
        {
          "locale": "en-US",
          "hreflang": "en-us",
          "enabled": true,
          "id": "e76bc4b0-d1dc-4e8b-85ff-7d8e463240a4"
        }
      ]
    }
  },
  "extensions": {
    "enableRefresh": "true"
  }
}
```

OR 

```json
{
  "data": {
    "allStrapiSiteConfig": {
      "nodes": [
        {
          "locale": "en-GB",
          "localizations": {
            "data": [
              {
                "locale": "en-US",
                "hreflang": "en-us",
                "enabled": true,
                "id": "e76bc4b0-d1dc-4e8b-85ff-7d8e463240a4"
              }
            ]
          },
          "hreflang": "en-gb",
          "enabled": true,
          "id": "3369956c-7e5b-5197-96cf-52a744bf9a90"
        }
      ]
    }
  },
  "extensions": {
    "enableRefresh": "true"
  }
}
```

**Comment by Razinsky** [comment]
Any updates here? I have the same issue:

This example that is presented in your GH page does not work for me when selecting a locale that is not the default. It works for default `en`.

```graphql
  # Get a single type in a specific locale
  strapiGlobal(locale: {eq: "fr"}) {
    locale
  }

  # ...yields null
  "data": {
    "strapiGlobal": null
  },
```

**Comment by remidej** [comment]
Please upgrade to the 2.0.0 stable version to have full support for localized single types. Once you do follow [the guide in the readme](https://github.com/strapi/gatsby-source-strapi#internationalization).

Basically you shouldn't try to populate the localizations field. Instead, set the locales you want in your gatsby config, then you can query your single type in all its localizations with using something like `allStrapiSingleType` in GraphQL

**Comment by Razinsky** [comment]
Thanks for answering promptly @remidej.

As a matter of fact, I'm using the `2.0.0` stable version and followed the guide in the readme with that bit. The result of a query with a locale filter e.g `locale: {eq: "fr"}` that is not the default yields `null`.

```
      pluginOptions: {
        i18n: {
          locale: 'all', // Fetch all localizations
        },
      },
```

**Comment by remidej** [comment]
What data do you get when you query `allStrapiGlobal`?

**Comment by Razinsky** [comment]
Querying
```graphql
allStrapiGlobal(filter: {locale: {eq: "fr"}}) {
    nodes {
      nodeName
    }
  }
```
yields
```graphql
"data": {
    "allStrapiGlobal": {
      "nodes": []
    }
  },
```

**Comment by Razinsky** [comment]
And to be clear, if I request `http://localhost:1337/api/global?locale=all` directly with Postman, I don't get other locales than default. So I don't even think it is related to the gatsby plugin 🤔

**Comment by Razinsky** [comment]
Pointing directly to any locale works via API `http://localhost:1337/api/global?locale=fr` where `fr` is not default. So I think there is maybe something happening with the `all` keyword.

**Comment by remidej** [comment]
`locale=all` doesn't work for single types on the Strapi rest API since it can't return arrays, so it's normal that your Postman request doesn't work.

But we now handle this in gatsby-source-strapi to allow users to get all locales anyway. We do this by making several requests under the hood. It was implemented in the PR #310 that was released in the stable v2.0.0

A couple of troubleshooting questions just in case:

- Are you sure you have content in Strapi for the `fr` locale?
- Are you sure the content is published and not a draft?
- Did you set pluginOptions on the right content type in your gatsby config?
- Did you restart the gatsby process after modifying your gatsby config?

Also, can you try using `locale: ['fr', 'en']` instead of `all`? Or whichever locales your app uses anyway. It's not something we recommend but I'm curious

**Comment by Razinsky** [comment]
Ok got it to work. I realized I was on `2.0.0-beta.0`. After updating to stable `2.0.0` and also updating `@strapi` from `4.1.1` to `4.1.7` it seems to work with filters 🙌 

Thank you for your support @remidej
